assets/javascripts/modules/mma/updatePayment.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import ReactDOM from 'react-dom';
+
+
+
+export const init = (elem) => {
+    let handler = window.StripeCheckout.configure(guardian.stripeCheckout)
+    ReactDOM.render(<Payment sub={elem.dataset.subId} email={elem.dataset.email} stripe={handler} />, elem)
+}
+
+const getDetails = async (url) => {
+    let resp = await fetch(`${url}/user-attributes/me/mma-paper`, { method: 'get', credentials: 'include' })
+    let json = await resp.json()
+    return json
+}
+const WAITING = 'waiting'
+const SUCCESS = 'success'
+const FAILURE = 'failure'
+const FORM = 'form'
+
+const CardUpdate = ({ card, handler }) => (<div><button className="button button--primary button--large" onClick={handler}>•••• •••• •••• {card.last4} Update Card</button></div>)
+const Success = () => (<p>Thank you, we have successfully updated your payment details.</p>)
+const Failure = () => (<p>Unfortunately, we are unable to update your payment details at this time, please contact the call centre.</p>)
+const Waiting = () => (<div className="loader js-loader is-loading">Processing&hellip;</div>)
+
+class Payment extends React.Component {
+
+    constructor(props) {
+        super(props)
+        const url = guardian.members_data_api
+
+        const token = (t) => {
+            let form = new FormData();
+            form.append('stripeToken', t.id)
+            form.append('publicKey', this.state.card.stripePublicKeyForUpdate)
```

Comment:
Does this take the AU/UK Stripe account changes into account? (https://github.com/guardian/members-data-api/pull/259)

Reply:
It should do, but I can't test it until we can sell subs through it.